### PR TITLE
List of references to Hail papers

### PR DESCRIPTION
**Pull request by gsarma**
- Started with Jon's informal list and added publications from Google Scholar by searching for our repo
- Checked article text to make sure I only included publications which made use of Hail or discussed it substantively and filtered out ones that only mention it parenthetically

@jbloom22 @tpoterba

### CODE REVIEW

File: hail/www/references.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
+
+# Scientific Output from the Hail Community  
+
+The following is a sample of research results that have used Hail as some part of the
+scientific pipeline.  If you would like your work added to this list, please
```
Comment (tpoterba) [inline hail/www/references.md]: can this be a link to the "edit this page on github" link?

**Comment by tpoterba** [comment]
can you open another PR to add yourself here:
https://github.com/hail-is/hail/blob/9239578c97b0af38d47070c986a4702eb87059e2/ci/ci/constants.py#L25

File: hail/www/references.md (hunk 1 above)
Comment (tpoterba) [reply hail/www/references.md]: can link directly here:

https://github.com/hail-is/hail/edit/master/hail/www/references.md

Comment (gsarma) [reply hail/www/references.md]: Yep, that's what I did.  Tried to find a way to make a local link, but didn't seem to be anything.  Squashed the dummy commits.

**Pull request closed without merging**